From de3209a2fbe50093236b95ec525eaea654e9dcb3 Mon Sep 17 00:00:00 2001
From: Vekhir <134215107+Vekhir@users.noreply.github.com>
Date: Sun, 18 Feb 2024 23:53:29 +0100
Subject: [PATCH] fix: Hide ambiguous instance for (!?)

base 4.19.0.0 introduces Data.List.!? which clashes with Data.Vector.!?, the latter of which is needed for vector operations.
---
 hledger-ui/Hledger/UI/AccountsScreen.hs | 5 +++++
 hledger-ui/Hledger/UI/RegisterScreen.hs | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/hledger-ui/Hledger/UI/AccountsScreen.hs b/hledger-ui/Hledger/UI/AccountsScreen.hs
index af5dd99bb9b..28361cce54e 100644
--- a/hledger-ui/Hledger/UI/AccountsScreen.hs
+++ b/hledger-ui/Hledger/UI/AccountsScreen.hs
@@ -1,5 +1,6 @@
 -- The accounts screen, showing accounts and balances like the CLI balance command.
 
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE TupleSections #-}
@@ -23,7 +24,11 @@ import Brick.Widgets.List
 import Brick.Widgets.Edit
 import Control.Monad
 import Control.Monad.IO.Class (liftIO)
+#if MIN_VERSION_base(4,19,0)
+import Data.List hiding (reverse, (!?))
+#else
 import Data.List hiding (reverse)
+#endif
 import Data.Maybe
 import qualified Data.Text as T
 import Data.Time.Calendar (Day)
diff --git a/hledger-ui/Hledger/UI/RegisterScreen.hs b/hledger-ui/Hledger/UI/RegisterScreen.hs
index d55d22d8252..92c2d7caefa 100644
--- a/hledger-ui/Hledger/UI/RegisterScreen.hs
+++ b/hledger-ui/Hledger/UI/RegisterScreen.hs
@@ -19,7 +19,11 @@ where
 import Control.Monad
 import Control.Monad.IO.Class (liftIO)
 import Data.Bifunctor (bimap, Bifunctor (second))
+#if MIN_VERSION_base(4,19,0)
+import Data.List hiding ((!?))
+#else
 import Data.List
+#endif
 import Data.Maybe
 import qualified Data.Text as T
 import qualified Data.Vector as V
